### PR TITLE
PV2-1600 - Apply refund to correct reporting funding line 

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.Application/Mapping/RequiredPaymentsProfile.cs [hunk 1]
```diff
@@ -206,8 +209,6 @@ public RequiredPaymentsProfile()
                 .Ignore(x => x.InstalmentAmount)
                 .Ignore(x => x.NumberOfInstalments)
                 .Ignore(x => x.LearningStartDate)
-                .Ignore(x => x.ApprenticeshipId)
-                .Ignore(x => x.ApprenticeshipPriceEpisodeId)
                 .Ignore(x => x.NumberOfInstalments)
                 .Ignore(x => x.ApprenticeshipEmployerType)
```
Comment (carlbm) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Mapping/RequiredPaymentsProfile.cs]: This will need to go

Comment (carlbm) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Mapping/RequiredPaymentsProfile.cs]: I actually have no idea what would happen here...

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/EarningEventProcessorBase.cs [hunk 2]
```diff
@@ -125,6 +123,8 @@ public async Task<ReadOnlyCollection<PeriodisedRequiredPaymentEvent>> HandleEarn
                         mapper.Map(earningEvent, requiredPaymentEvent);
                         mapper.Map(requiredPayment, requiredPaymentEvent);
 
+                        requiredPaymentEvent.ApprenticeshipEmployerType = requiredPayment.ApprenticeshipEmployerType;
```
Comment (carlbm) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Processors/EarningEventProcessorBase.cs]: I think that we should do a check. If the required payment has an `Amount < 0` then apply this line.
If positive then the properties are set in datalocks - but this method is used for all payments, including ACT2 etc. Leaving this here risks an ACT2 learner who used to be ACT1 etc being incorrectly funded.

Comment (carlbm) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Processors/EarningEventProcessorBase.cs]: Also, we should include `ApprenticeshipId` and `ApprenticeshipPriceEpisodeId` in the copy

File: src/SFA.DAS.Payments.RequiredPayments.Domain/Services/RefundService.cs [hunk 3]
```diff
@@ -43,8 +47,10 @@ public List<RequiredPayment> GetRefund(decimal amount, List<Payment> paymentHist
                         SfaContributionPercentage = group.Key.SfaContributionPercentage,
                         PriceEpisodeIdentifier = group.Key.PriceEpisodeIdentifier,
                         AccountId = group.Key.AccountId,
-                        TransferSenderAccountId = group.Key.TransferSenderAccountId
-
+                        TransferSenderAccountId = group.Key.TransferSenderAccountId,
+                        ApprenticeshipEmployerType = group.Key.ApprenticeshipEmployerType,
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.RequiredPayments.Domain/Services/RefundService.cs]: I had a look at the GetPaymentHistory Method and we don't seem to be loading (ApprenticeshipEmployerType) from the Database.  I think we need additional code changes to load this field from DB.

File: src/SFA.DAS.Payments.RequiredPayments.Application/Mapping/RequiredPaymentsProfile.cs (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Mapping/RequiredPaymentsProfile.cs]: The `Ignore` overrides the `Map`

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/EarningEventProcessorBase.cs (hunk 2 above)
Comment (benagain) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Processors/EarningEventProcessorBase.cs]: Added `AddRefundCommitmentDetails`

File: src/SFA.DAS.Payments.RequiredPayments.Domain/Services/RefundService.cs (hunk 3 above)
Comment (benagain) [reply src/SFA.DAS.Payments.RequiredPayments.Domain/Services/RefundService.cs]: You're right (of course 😄) - I've made `GetPaymentHistory` fill in the commitment properties too now.

**Pull request merged**